File: app/src/main/java/com/rosebay/odds/dagger/module/ContextModule.java
package com.rosebay.odds.dagger.module;

import android.content.Context;

import com.rosebay.odds.dagger.component.OddsApplicationScope;

import javax.annotation.Nonnull;

import dagger.Module;
import dagger.Provides;

@Module
public class ContextModule {

    private final Context context;

    public ContextModule(@Nonnull Context context) {
        this.context = context;
    }

    @Provides
    @OddsApplicationScope
    public Context providesContext(){
        return context;
    }

}

File: app/src/main/java/com/rosebay/odds/dagger/module/ContextModule.kt
package com.rosebay.odds.dagger.module

import android.content.Context

import com.rosebay.odds.dagger.component.OddsApplicationScope

import dagger.Module
import dagger.Provides

@Module
open class ContextModule(private val context: Context) {

    @Provides
    @OddsApplicationScope
    open fun providesContext(): Context = context
}